src/libData/AccountData/AccountStoreSC.tpp

```diff
@@ -1694,9 +1701,12 @@ bool AccountStoreSC<MAP>::ParseCallContractJsonOutput(
         return false;
       }
 
-      // prepare IPC with the recipient contract address
-      m_scillaIPCServer->setContractAddressVerRoot(recipient, scilla_version,
-                                                   account->GetStorageRoot());
+      // prepare IPC with current blockchain info provider.
+      auto sbcip1 = std::make_unique<ScillaBCInfo>(
+          m_curBlockNum, m_curDSBlockNum, m_originAddr, recipient,
+          account->GetStorageRoot(), scilla_version);
+      m_scillaIPCServer->setBCInfoProvider(std::move(sbcip1));
+
```

Comment:
So, is my understanding correct: this is the first PR towards enabling the EVM to access the block info (so it does not currently), but scilla can do so already?

I understand why this is added in UpdateAccounts, but not ParseCallContractJsonOutput function (and why it is set twice - does the storage root change between calls?). My reading of the code is that ParseCallContractJsonOutput is called once the VM is finished executing, so why does it need the block info etc. to be attached?

Reply:
EVM needs more blockchain info than Scilla as of 8.2.  Scillla in 8.3 .. old master needs more hence they added this BC Info provider, which I just extended.

Spot on re ParseContractJsonOutput - it just really rebased the change first, and changes related to Scilla also got in.
I actually overlooked this. But it indeed must get in, because if we change the way this gets handled for fetchBlockchainInfo method, it also affects Scilla, so the object must be in place.

Why does `ParseCallContractJsonOutput` call Scilla again? Well, because there is recursion there - Scilla must be able to call other smart contracts.  In fact, I think this recursion is not needed.  Scilla only allows tail calls to other smart contracts - meaning this should be a loop rather than recursion.  File https://zilliqa-jira.atlassian.net/browse/ZIL-4650